components/ModelSelect.tsx

```
@@ -8,27 +11,121 @@ interface Props {
   const bg = isDark ? 'bg-[#1A1B26]' : 'bg-[#fff]';
   const textColor = isDark ? 'text-neutral-200 ' : 'text-black';
 
+  if (loading) {
+    return (
+      <div className={`${bg} ${textColor} h-[40px] w-fit rounded-md px-4 py-2`}>
+        Loading models...
+      </div>
+    );
+  }
+
+  const availableModels = useMemo(() => {
```

ESLint

[error] React Hook "useMemo" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?

(react-hooks/rules-of-hooks)